sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -3345,39 +3345,51 @@ case class FormatNumber(x: Expression, d: Expression)
   group = "string_funcs")
 case class Sentences(
     str: Expression,
-    language: Expression = Literal(""),
-    country: Expression = Literal(""))
-  extends TernaryExpression with ImplicitCastInputTypes with CodegenFallback {
+    language: Option[Expression] = None,
+    country: Option[Expression] = None)
+  extends Expression with ImplicitCastInputTypes {
+
+  def this(str: Expression) = this(str, None, None)
 
-  def this(str: Expression) = this(str, Literal(""), Literal(""))
-  def this(str: Expression, language: Expression) = this(str, language, Literal(""))
```

Comment:
This change can break existing user apps. I believe we should avoid such kind of changes even in major release. 

How about to correct the function description, and allow 1, 2 and 3 arguments? It seems there is no restriction from implementation perspective, so, you can construct `new Locale(languageStr.toString)`

Reply:
Sorry, I missed this message.
Do we need to add function like this (`def sentences(string: Column, language: Column): Column`) in `functions.scala`? This looks consistent.

Reply:
Indeed, both of the following forms can construct `Locale`:
- new Locale(languageStr.toString) (when parameter `language` is only available)
- new Locale(languageStr.toString, countryStr.toString) (when parameter `language` and `country` are all  available)